onadata/apps/logger/migrations/0019_purge_deleted_instances.py

```diff
@@ -0,0 +1,42 @@
+# Generated by Django 2.2.14 on 2021-06-30 19:00
+from django.conf import settings
+from django.db import migrations
+
+
+def purge_deleted_instances(apps, schema_editor):
+    """
+    Remove all submissions that have been already marked as deleted from both
+    PostgreSQL and MongoDB. If this is too slow, revert to a previous release
+    and run the code in
+    https://github.com/kobotoolbox/kobocat/issues/696#issuecomment-809622367
+    using `manage.py shell_plus`.
+    """
+    Instance = apps.get_model('logger', 'Instance')
+    to_purge = Instance.objects.exclude(deleted_at=None).only('pk')
+    if not to_purge.exists():
+        return
+    print(
+        f'Purging {to_purge.count()} deleted instances...', end='', flush=True
+    )
```

Comment:
FYI: I cannot see the output of these `print` .
In other migrations, we used `sys.stderr.write`

Reply:
Oops, I'm only now seeing this! I'm surprised you couldn't see it the output if count was greater than zero on your database, but `print()` certainly doesn't seem ideal to use here (although neither does `sys.stdout.write` :wink:). I guess migrations are part of a management command and theoretically ought to have access somehow to `self.stdout` [provided somewhere by](https://docs.djangoproject.com/en/2.2/howto/custom-management-commands/) the `BaseCommand` class.

Be more insistent that I acknowledge your feedback next time :stuck_out_tongue: